cranelift/codegen/src/isa/aarch64/lower/isle.rs

```diff
@@ -335,7 +335,10 @@ where
     }
 
     fn shift_mask(&mut self, ty: Type) -> ImmLogic {
-        let mask = (ty.bits() - 1) as u64;
+        let bits = ty.lane_bits();
+        debug_assert_eq!((bits & (bits - 1)), 0, "Type size not power of 2");
```

Comment:
There is a `.is_power_of_two()` method on the integer types that makes this clearer, IMHO (internally I suspect it uses exactly this idiom).